dds/DCPS/InstanceState.h

```diff
@@ -131,6 +131,12 @@ class OpenDDS_Dcps_Export InstanceState : public ACE_Event_Handler {
 
   DDS::InstanceHandle_t instance_handle() const { return handle_; }
 
+  /// Return string of the name of the current instance state
+  std::string instance_state_string() { return instance_state_string(instance_state_); }
```

Comment:
This can be a const member function.
See PoolAllocator.h for a safety-profile-compatible way of using std::string.